policy-controller/k8s/index/src/outbound/index.rs

```diff
@@ -651,9 +660,16 @@ impl Index {
         }
     }
 
-    fn reinitialize_egress_watches(&mut self) {
+    fn reinitialize_egress_watches(&mut self, namespace: Option<Arc<String>>) {
```

Comment:
This is an optimization I knew could be made. The reason why I included it in this PR is because it minimises the flakiness significantly. As these tests are namespace-local, having modifications to egress networks in one namespace force reinitialize the watches for other namespaces leads to flakiness in these

Reply:
This change makes it so streams that have resolved to `EgressNetwork` parents that live in namespace A are restarted when a modification to an `EgressNetwork` in namespace B happens. This is an optimization that results in a more correct behavior. The test flakiness is caused as I have pointed out by observing the above mentioned scenario each test is scoped to a namespace bu modifications to egress networks in other namespace (other tests) cause stream restarts, etc which causes flakiness.